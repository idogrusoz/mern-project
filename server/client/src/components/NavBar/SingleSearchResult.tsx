import { Avatar, Grid, makeStyles, Typography } from "@material-ui/core";
import React, { FunctionComponent } from "react";
import { SearchedUser } from "../../../../interfaces/user.interfaces";

const useStyles = makeStyles({
    result: {
        marginTop: 10,
        "&:hover": {
            backgroundColor: "#ffffff",
            cursor: "pointer",
        },
    },
});

type SingleSearchResultProps = {
    user: SearchedUser;
};

const SingleSearchResult: FunctionComponent<SingleSearchResultProps> = ({ user }) => {
    const classes = useStyles();
    return (
        <Grid container alignItems="center" style={{ marginTop: 10 }} className={classes.result}>
            <Grid item xs={3}>
                <Avatar src={user.image} onClick={() => console.log(user._id)} />
            </Grid>
            <Grid item xs={9}>
                <Typography onClick={() => console.log(user._id)}>{user.userName}</Typography>
            </Grid>
        </Grid>
    );
};

export default SingleSearchResult;
