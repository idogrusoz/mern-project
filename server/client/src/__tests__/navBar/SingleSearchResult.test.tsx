import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import SingleSearchResult, { SingleSearchResultProps } from "../../components/NavBar/SingleSearchResult";

describe("SingleSearchResult test", () => {
    const props: SingleSearchResultProps = {
        user: {
            _id: "id",
            displayName: "displayName",
            image: "",
            userName: "userName",
            followers: [],
            following: [],
        },
        pickUser: () => {},
    };
    const wrapper = shallow(<SingleSearchResult {...props} />);
    it("renders without crash", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
