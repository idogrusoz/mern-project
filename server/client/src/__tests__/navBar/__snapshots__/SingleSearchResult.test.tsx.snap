// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSearchResult test renders without crash 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="makeStyles-result-1"
  container={true}
  onClick={[Function]}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={3}
  >
    <WithStyles(ForwardRef(Avatar))
      onClick={[Function]}
      src=""
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={9}
  >
    <WithStyles(ForwardRef(Typography))
      onClick={[Function]}
    >
      userName
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
